From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Taiyou06 <kaandindar21@gmail.com>
Date: Thu, 8 May 2025 00:05:01 +0200
Subject: [PATCH] Save world async properly

P.S from Tai: I've been using this fix for weeks in my own server but didn't had balls to push it as thought it may cause issues but, it's merged in paper 1.21.5 now.

diff --git a/net/minecraft/server/level/ServerLevel.java b/net/minecraft/server/level/ServerLevel.java
index cb18e64db6c1eeb1b4ac920b606a682bce5db8fa..480761c5bc08f8587f5e37ceef22678bf4e7c691 100644
--- a/net/minecraft/server/level/ServerLevel.java
+++ b/net/minecraft/server/level/ServerLevel.java
@@ -1577,7 +1577,7 @@ public class ServerLevel extends Level implements ServerEntityGetter, WorldGenLe
         }
 
         if (doFull) {
-            this.saveLevelData(true);
+            this.saveLevelData(false);
         }
         // chunk autosave is already called by the ChunkSystem during unload processing (ChunkMap#processUnloads)
         // Copied from save()
